pkg/app/piped/cloudprovider/kubernetes/diff.go

```diff
@@ -75,6 +84,56 @@ func DiffList(olds, news []Manifest, opts ...diff.Option) (*DiffListResult, erro
 	return cr, nil
 }
 
+// If at least one of old and new is nil or not secret, then return new, nil
+func normalizeNewSecret(old, new *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+	if old == nil || new == nil {
+		return new, nil
+	}
+
+	gvkOld := old.GroupVersionKind()
+	gvkNew := new.GroupVersionKind()
+
+	if gvkOld.Kind != "Secret" || gvkNew.Kind != "Secret" {
+		return new, nil
+	}
+
```

Comment:
Btw, I think we should place this check on the caller's side.
This function is expected to accept only Secret manifests since the function name is `...Secret`.

Reply:
Thank you! I'll move this!